src/object/C4Shape.cpp

```diff
@@ -273,250 +331,346 @@ bool C4Shape::LineConnect(int32_t tx, int32_t ty, int32_t cvtx, int32_t ld, int3
 		int32_t cix;
 		int32_t ciy;
 		for (int irange = 4; irange <= 12; irange += 4)
+		{
 			for (cix = ix - irange / 2; cix <= ix + irange; cix += irange)
+			{
 				for (ciy = iy - irange / 2; ciy <= iy + irange; ciy += irange)
 				{
-					if (PathFree(cix,ciy,tx,ty) && PathFree(cix,ciy,VtxX[cvtx+ld],VtxY[cvtx+ld]))
+					if (PathFree(cix, ciy, tx, ty) && PathFree(cix, ciy, VtxX[cvtx + ld], VtxY[cvtx + ld]))
 					{
 						found = true;
 						goto out;
 					}
 				}
+			}
+		}
 out:
 		if (!found)
 		{
-			// try bending directly at path the line took
-			// allow going through vehicle in this case to allow lines through castles and elevator shafts
+			// Try bending directly at path the line took
+			// Allow going through vehicle in this case to allow lines through castles and elevator shafts
 			cix = oldx;
 			ciy = oldy;
-			if (!PathFreeIgnoreVehicle(cix,ciy,tx,ty) || !PathFreeIgnoreVehicle(cix,ciy,VtxX[cvtx+ld],VtxY[cvtx+ld]))
-				if (!PathFreeIgnoreVehicle(cix,ciy,tx,ty) || !PathFreeIgnoreVehicle(cix,ciy,VtxX[cvtx+ld],VtxY[cvtx+ld]))
```

Comment:
What's even happening here? Isn't this just the same test done twice?

Reply:
At first I thought that it wants to get certain coordinates, but those are the 5th and 6th parameters and they are unused here. 
It seems that this actually does nothing then. I can remove the double check and see if it breaks something.